features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/TryOut/TryOutConsole.jsx

```diff
@@ -96,8 +102,11 @@ const TryOutConsole = () => {
         if (selectedDeployment && oasDefinition) {
             const selectedGWEnvironment = publisherSettings.environment
                 .find((env) => env.name === selectedDeployment.name);
-            const selectedDeploymentVhost = selectedGWEnvironment.vhosts
+            let selectedDeploymentVhost = selectedGWEnvironment.vhosts
                 .find((vhost) => vhost.host === selectedDeployment.vhost);
+            if (!selectedDeploymentVhost) {
+                selectedDeploymentVhost = { ...CONSTS.DEFAULT_VHOST, host: selectedDeployment.vhost };
```

Comment:
I think this `host` property will always have `undefined` as its value

Reply:
Yes, this handles the scenario, in issue https://github.com/wso2/product-apim/issues/10711

1. Create a dynamic env from the admin portal (us-region)
1. Create two vhosts: us1.wso2.com, us2.wso2.com from the admin portal
1. Deploy an API with us2.wso2.com in the publisher portal
1. Delete vhost us2.wso2.com from the admin portal
1. Tryout API within publisher portal with the created env (us-region)
1. Crash Tryout page

So there is no vhost `us2.wso2.com`, but an revision is deployed with `us2.wso2.com`.